Sources/Sentry/SentryMetricProfiler.mm

```diff
@@ -156,12 +159,16 @@ - (void)stop
 - (void)registerSampler
 {
     __weak auto weakSelf = self;
-    _timer = [_timerWrapper scheduledTimerWithTimeInterval:kSentryMetricProfilerTimeseriesInterval
-                                                   repeats:YES
-                                                     block:^(NSTimer *_Nonnull timer) {
-                                                         [weakSelf recordCPUPercentagePerCore];
-                                                         [weakSelf recordMemoryFootprint];
-                                                     }];
+    _timer =
+        [_dispatchFactory sourceWithInterval:intervalNs
+                                      leeway:(uint64_t)1e9 / frequencyHz
+                                   queueName:"io.sentry.metric-profiler"
+                                  attributes:dispatch_queue_attr_make_with_qos_class(
+                                                 DISPATCH_QUEUE_CONCURRENT, QOS_CLASS_UTILITY, 0)
+                                eventHandler:^{
+                                    [weakSelf recordCPUPercentagePerCore];
+                                    [weakSelf recordMemoryFootprint];
```

Comment:
`recordCPUPercentagePerCore` and `recordMemoryFootprint` were previously called on the main thread, are they thread safe?

Reply:
Good question. Their calls to `task_info` and `host_processor_info` are not synchronized, but their modifications of the data structure those readings are stored in are inside `@synchronized` blocks. Is this safe, or do you think we should move the `task_info`/`host_processor_info` calls inside the synchronization as well?

Reply:
i improved the method that reads from the data structure as well in https://github.com/getsentry/sentry-cocoa/pull/2956/commits/adef4bc78ea2111056102e91a9a8175b0ceece29